lib/govuk_ab_testing/requested_variant.rb

```diff
@@ -0,0 +1,45 @@
+module GovukAbTesting
+  class RequestedVariant
+    attr_reader :ab_test, :request
+
+    # @param ab_test_name [AbTest] Lowercase A/B test name, like `example`
+    # @param request [ApplicationController::Request] the `request` in the
+    # controller.
+    def initialize(ab_test, request)
+      @ab_test = ab_test
+      @request = request
+    end
+
+    # Get the bucket this user is in
+    #
+    # @return [String] the current variant, "A" or "B"
+    def variant_name
+      request.headers[ab_test.request_header] == "B" ? "B" : "A"
+    end
+
+    # @return [Boolean] if the user is to be served variant A
+    def variant_a?
```

Comment:
Ah fair enough - in that case I agree it's better to have more descriptive names.